opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,496 @@
+
+    @staticmethod
+    def get_opensearch_domain_info(region, domain_name):
+        """
+        Retrieve the OpenSearch domain endpoint and ARN based on the domain name and region.
+
+        :param region: AWS region.
+        :param domain_name: Name of the OpenSearch domain.
+        :return: Tuple containing domain endpoint and ARN.
+        """
+        try:
+            opensearch_client = boto3.client('opensearch', region_name=region)
+            response = opensearch_client.describe_domain(DomainName=domain_name)
+            domain_status = response['DomainStatus']
+            domain_endpoint = domain_status.get('Endpoint')
+            domain_arn = domain_status['ARN']
+            return domain_endpoint, domain_arn
+        except Exception as e:
+            print(f"Error retrieving OpenSearch domain info: {e}")
+            return None, None
+
+    def get_ml_auth(self, create_connector_role_name):
+        """
+        Obtain AWS4Auth credentials for ML API calls using the specified IAM role.
+
+        :param create_connector_role_name: Name of the IAM role to assume.
+        :return: AWS4Auth object with temporary credentials.
+        """
+        create_connector_role_arn = self.iam_helper.get_role_arn(create_connector_role_name)
+        if not create_connector_role_arn:
+            raise Exception(f"IAM role '{create_connector_role_name}' not found.")
+
+        temp_credentials = self.iam_helper.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+        return awsauth
+
+    def create_connector(self, create_connector_role_name, payload):
+        """
+        Create a connector in OpenSearch using the specified role and payload.
+
+        :param create_connector_role_name: Name of the IAM role to assume.
+        :param payload: Payload data for creating the connector.
+        :return: ID of the created connector.
+        """
+        create_connector_role_arn = self.iam_helper.get_role_arn(create_connector_role_name)
+        temp_credentials = self.iam_helper.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+
+        path = '/_plugins/_ml/connectors/_create'
+        url = self.opensearch_domain_url + path
+
+        headers = {"Content-Type": "application/json"}
+
+        response = requests.post(url, auth=awsauth, json=payload, headers=headers)
+        print(response.text)
+        connector_id = json.loads(response.text).get('connector_id')
+        return connector_id
+
+    def get_task(self, task_id, create_connector_role_name):
```

Comment:
can we re-use this method: https://github.com/opensearch-project/opensearch-py-ml/blob/39284d6e99cd9b449bcfe99511c68a63d897ca78/opensearch_py_ml/ml_commons/ml_commons_client.py#L389 ?

Reply:
Yes we can, I have leveraged that method in my latest commit, thank you